### PR TITLE
Improvement in messageActionSheet

### PR DESCRIPTION
**Pull request by vishwesh3**
#651

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -40,6 +40,7 @@
     "react-native-fabric": "^0.4.1",
     "react-native-safari-view": "^2.0.0",
     "react-native-scrollable-tab-view": "^0.7.0",
+    "react-native-simple-toast": "0.0.6",
```
Comment (vishwesh3) [inline package.json]: This uses scalessec/Toast on iOS and the standard ToastAndroid on Android.

**Comment by vishwesh3** [comment]
Completed 👍 
Ready for review.

File: src/common/showToast.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import Toast from 'react-native-simple-toast';
```
Comment (borisyankov) [inline src/common/showToast.js]: This likely belongs in utils.
If you are going to have more functions related to toast call it 'toats.js' if not export default.

File: src/common/showToast.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
...
+
+export const showToast = (message) => Toast.show(message, Toast.SHORT);
```
Comment (borisyankov) [inline src/common/showToast.js]: We will need the message translated. Maybe here is the best way to handle that?

File: src/message/__tests__/messageActionSheet-test.js [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    const buttons = constructActionButtons({ message, auth, narrow, subscriptions, mute });
+
+    expect(buttons).toContain('Narrow to conversation');
+  });
+
+  test('do not show narrow option in private narrow', () => {
+    narrow = privateNarrow('abc@zulip.com');
+    const buttons = constructActionButtons({ message, auth, narrow, subscriptions, mute });
+
+    expect(buttons).not.toContain('Narrow to conversation');
```
Comment (borisyankov) [inline src/message/__tests__/messageActionSheet-test.js]: I think the 'Narrow to conversation' tests are a bit too repetitive. Can you combine them into one?

File: src/utils/narrow.js [hunk 5]
```diff
@@ -113,9 +113,10 @@ export const canSendToNarrow = (narrow) =>
   isStreamNarrow(narrow) ||
   isTopicNarrow(narrow);
 
-export const narrowFromMessage = (message) => {
+export const narrowFromMessage = (message, auth) => {
```
Comment (borisyankov) [inline src/utils/narrow.js]: If you need just the email, do pass only the email.

File: src/common/showToast.js (hunk 2 above)
Comment (vishwesh3) [inline src/common/showToast.js]: Moved it to `utils.`
In future we can add more functions. For now one is sufficient.

File: src/common/showToast.js (hunk 3 above)
Comment (vishwesh3) [inline src/common/showToast.js]: Got stuck here. How to get translated strings in this function?

File: src/message/__tests__/messageActionSheet-test.js (hunk 4 above)
Comment (vishwesh3) [inline src/message/__tests__/messageActionSheet-test.js]: Combined to 1 test 👍

File: src/utils/narrow.js (hunk 5 above)
Comment (vishwesh3) [inline src/utils/narrow.js]: done 👍

**Comment by vishwesh3** [comment]
Updated branch 👍 
also posted a question [here](https://github.com/zulip/zulip-mobile/pull/658#discussion_r120321447)

**Comment by vishwesh3** [comment]
@borisyankov rebased 👍 
this is also ready to go :runner:

**Comment by borisyankov** [comment]
Looks good now!

**Pull request merged**